### DISCUSSION TITLE
Crash of plantuml on MAC

### DISCUSSION THREAD

**Discussion by mjaentsch** [plantuml/plantuml#1581] [Q&A] [answered]
I installed plantuml on Mac recently using brew now running PlantUML version 1.2023.11 and OpenJDK Runtime Environment (64-Bit Server VM). Even very simple plantumls cause a crash in the Java Runtime Environment:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x0000000105c26fc8, pid=16469, tid=9219
#
# JRE version: OpenJDK Runtime Environment Homebrew (21.0) (build 21)
# Java VM: OpenJDK 64-Bit Server VM Homebrew (21, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-aarch64)
# Problematic frame:
# C  [libawt.dylib+0x2efc8]  fillAAPgram+0x3e0
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# ~/hs_err_pid16469.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/Homebrew/homebrew-core/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
zsh: abort      plantuml example.puml
````

It doesn't actually matter which diagram I am trying to convert. There is an even simpler example which converts just fine:

@startuml

'skinparam monochrome true

Rectangle "Abstraction" as a <<Middleware>> {
	Rectangle "Data of Sensor Objects" as dso
	Rectangle "Configuration Data" as ioco
}

Rectangle "Application Level" as al <<pure>> {
	Rectangle "Configuration Data" as aco
	Rectangle "Application Data Objects" as ado
}

al -[hidden]d- ioa

@enduml

The following does not convert with the same error:

@startuml

Component "test"

@enduml

Does anyone have any idea?

**Comment by mjaentsch** [comment]
I find it a bit weird that I seem to be the only one having this issue, since it seems so basic. Does anyone have any pointers or ideas what to try?

**Comment by mjaentsch** [comment]
So I resolved by installing oracle-jdk instead. Didn't try any of the other ones but this at least worked. Since plantuml from brew depends on openjdk it cannot be used anymore but it didn't do much other than put the jar somewhere and a bash script to launch java. Nothing that can't be easily replicated. 

Just in case someone stumbles across this.

**Reply by DWochnik** [reply]
Same for me, but thanks a million for the workaround!

It crashs with
openjdk version "21.0.1" 2023-10-17
OpenJDK Runtime Environment Homebrew (build 21.0.1)
OpenJDK 64-Bit Server VM Homebrew (build 21.0.1, mixed mode, sharing)

but works with
java version "17.0.9" 2023-10-17 LTS
Java(TM) SE Runtime Environment (build 17.0.9+11-LTS-201)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.9+11-LTS-201, mixed mode, sharing)

Perhaps it is more a compatibility problem with the Java version? Either way, I'm just happy that it works and haven't looked at it any further.

**Comment by c835722** [comment]
Yes, so also getting this and it's a blocking issue for me. 

MacOS 14.1, 
OpenJDK Runtime Environment Homebrew (21.0.1) (build 21.0.1), 
PlantUML version 1.2023.12 (Sat Oct 21 00:49:54 AEDT 2023)
(GPL source distribution)
Java Runtime: OpenJDK Runtime Environment
JVM: OpenJDK 64-Bit Server VM
Default Encoding: UTF-8
Language: en
Country: AU
 PLANTUML_LIMIT_SIZE: 4096
Dot version: dot - graphviz version 9.0.0 (20230911.1827)

**Comment by jsy1001** [comment]
I'm also getting this crash.

➜ plantuml --version
PlantUML version 1.2023.12 (Fri Oct 20 14:49:54 BST 2023)
(GPL source distribution)
Java Runtime: OpenJDK Runtime Environment
JVM: OpenJDK 64-Bit Server VM
Default Encoding: UTF-8
Language: en
Country: GB

PLANTUML_LIMIT_SIZE: 4096

Dot version: dot - graphviz version 9.0.0 (20230911.1827)
Installation seems OK. File generation OK

**Comment by schspa** [comment]
I have same issues, resolved by using oracle-jdk with `brew install --cask oracle-jdk`

**Comment by c835722** [comment, marked as answer]
The upstream brew package appears fixed for me now and I am unable to reproduce it.
If there are no updates here in 1 week, we will close this item.

**Comment by mjaentsch** [comment]
I can confirm it works for me now. Thanks!
